tests/ui/abi/compatibility.rs

```diff
@@ -37,9 +37,11 @@
 // revisions: wasi
 //[wasi] compile-flags: --target wasm32-wasi
 //[wasi] needs-llvm-components: webassembly
-// revisions: nvptx64
-//[nvptx64] compile-flags: --target nvptx64-nvidia-cuda
-//[nvptx64] needs-llvm-components: nvptx
+// FIXME: disabled on nvptx64 since the target ABI fails the sanity check
+/* revisions: nvptx64
+  [nvptx64] compile-flags: --target nvptx64-nvidia-cuda
+  [nvptx64] needs-llvm-components: nvptx
+*/
```

Comment:
@kjetilkjeka I had to disable this test on nvptx64 since the `extern "C"` ABI uses `Direct` pass mode in invalid ways. I think that's caused by this code:

https://github.com/rust-lang/rust/blob/b75b3b3afecfb6638c7c75ad665f8f3091aab1ae/compiler/rustc_target/src/abi/call/nvptx64.rs#L10-L14

That's invalid since it doesn't say what to do for smaller aggregates, and they default to the (bad) `Direct`. Instead you have to say which register they are supposed to be passed in. You can check what the other targets are doing (nvptx is the only target that fails this assertion).